### PR TITLE
fix: improve v2 network support

### PR DESCRIPTION
**Pull request by JFrankfurt**

### CODE REVIEW

File: src/constants/chains.ts [hunk 1]
```diff
@@ -41,16 +41,9 @@ export const SUPPORTED_GAS_ESTIMATE_CHAIN_IDS = [
...
  */
-export const UNSUPPORTED_V2POOL_CHAIN_IDS = [
-  ChainId.POLYGON,
-  ChainId.OPTIMISM,
-  ChainId.ARBITRUM_ONE,
-  ChainId.BNB,
-  ChainId.ARBITRUM_GOERLI,
-  ChainId.AVALANCHE,
-] as const
+export const SUPPORTED_V2POOL_CHAIN_IDS = [ChainId.MAINNET, ChainId.GOERLI] as const
```
Comment (tinaszheng) [inline src/constants/chains.ts:46]: sepolia works too!

**Review by tinaszheng — APPROVED** [review]
thoughts on also amending `useV2RouterContract` to return undefined if the chainId is not supported?

File: src/pages/Pool/index.tsx [hunk 2]
```diff
@@ -225,7 +226,6 @@ export default function Pool() {
   }
 
   const showConnectAWallet = Boolean(!account)
-  const showV2Features = Boolean(V2_FACTORY_ADDRESSES[chainId])
```
Comment (zzmp) [inline src/pages/Pool/index.tsx:228]: My only issue here is that you're adding a new file to the V2 checking codepath, which complicates adding add'l chains. However, I don't think we plan on adding add'l chains to V2, so this seems fine.

**Review by zzmp — APPROVED** [review]

**Review by vm — APPROVED** [review]
just approving after main was merged

**Pull request merged**